backend/kernelCI_app/views/hardwareDetailsView.py

```diff
@@ -94,6 +111,13 @@ def sanitize_records(self, records):
                 )
                 tests_or_boots["failReasons"][extract_error_message(r["misc"])] += 1
 
+            archKey = r["build_architecture"] + r["build_compiler"]
```

Comment:
`build_architecture` and `build_compiler` fields can be null. This throws a error in that situation